controllers/welcome.rb

```diff
@@ -1,3 +1,89 @@
+# ===== welcome
 get '/' do
+  @apartments = Apartment.all
   erb :welcome
 end
+
+# ===== ===== ===== APARTMENTS ===== ===== =====
+# ===== show apartment
+get '/apartments/:id' do
+  @apartment = Apartment.find(@params[:id])
+  erb :apartment/show
+end
+
+# ===== edit apartment
+get '/apartments/:id/edit' do
+  @apartment = Apartment.find(@params[:id])
+  erb :apartment/edit
+end
+
+# ===== update apartment
+put '/apartments/:id/update' do
+  @apartment = Apartment.find(@params[:id])
+  @apartment.update(address: params[:address], monthly_rent: params[:monthly_rent], sqft: params[:sqft], num_beds: params[:num_beds], num_baths: params[:num_baths])
+  redirect "/apartments"
+end
+
+# ===== new apartment
+get '/apartments/new' do
```

Comment:
This route probably needs to be above the `show` route. That's because a GET request to "/apartments/new" will match the show route first... (Sinatra will treat the word `new` as the `: id` in the show route.